### PR TITLE
Makes Emitter locks silicon compatible

### PR DESCRIPTION
**Pull request by Wayland-Smithy**
## About The Pull Request

Lock proc and callers.

## Why It's Good For The Game

AI and Borgs do not bypass the lock and currently have no way to disable it.

## Changelog
:cl:
fix: Silicons can now toggle emitter locks with secondary attack.
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
Isn't this an intentional balance thing? Or was that loading the rad collectors... 🤔

**Comment by Wayland-Smithy** [comment]
> loading the rad collectors

Correct someone with hands must load the collectors, also if it was intentional it's not great because emitters can only be locked **on** and thus could prevent the AI from saving the SM when it's mix is sabotaged or borgs even working around them safely.

**Comment by Time-Green** [comment]
This seems like a balance change. Pretty sure the locking mechanism is exactly for the purpose of keeping meddlers out, including silicons. And even if they aren’t, this has been in the game for so long that it’s been generally accepted

An engineering borg can always cut the wires underneath to turn it off, or just beat the crap out of it if lifes depend on it

**Comment by Wayland-Smithy** [comment]
"Don't change this because that is the way it's always been" is just the worst argument I could imagine for something like this. The only meddling this prevents from AI or Borgs is quick emitter shutdowns which I'm sure the inability to do has led to many preventable delaminations.

**Comment by newfren** [comment]
This was originally in place to stop the ai from being able to release the singularity, as emitters being "on" was the safe state for that engine. Given that emitters being "off" is the safe state for the supermatter, this change seems to make sense?

**Comment by Time-Green** [comment]
> "Don't change this because that is the way it's always been" is just the worst argument I could imagine for something like this. The only meddling this prevents from AI or Borgs is quick emitter shutdowns which I'm sure the inability to do has led to many preventable delaminations.

My issue is mostly with you calling a balance change a fix

**Comment by Wayland-Smithy** [comment]
The Maintainers are free to re-label/changelog it, but I don't think it impacts balance in the slightest.

**Comment by 81Denton** [comment]
This isn't an issue IMO since we no longer have singulo/tesla containment fields to deal with

File: code/modules/power/singularity/emitter.dm [hunk 1]
```diff
@@ -195,6 +195,10 @@
 	if(. && !anchored)
 		step(src, get_dir(user, src))
 
+/obj/machinery/power/emitter/attack_ai_secondary(mob/user, list/modifiers)
```
Comment (Mothblocks) [inline code/modules/power/singularity/emitter.dm:198]: Looks like this is only the AI. Are you sure this works with cyborgs?

Comment (Wayland-Smithy) [reply code/modules/power/singularity/emitter.dm:198]: Yeah borgs call this proc unless theirs is overridden:
```
/atom/proc/attack_robot_secondary(mob/user, list/modifiers)
	return attack_ai_secondary(user, modifiers)
```

Comment (Mothblocks) [reply code/modules/power/singularity/emitter.dm:198]: It's not that I don't think you didn't test it, but it's that it's very common for edge cases to not come up in personal playtesting.

Comment (Wayland-Smithy) [reply code/modules/power/singularity/emitter.dm:198]: Just want to clear up all doubts. No offense intended power mover blocks.
